client/src/views/Home/index.tsx

```diff
@@ -4,51 +4,66 @@ import { AppButton, Select } from "../../components";
 import { useAuth } from "../../hooks";
 import "./Home.scoped.scss";
 
+const buildItemsForSelect = (arr: { id?: string, name: string }[]) => (arr || []).map(({ name: label, id: value }) => ({ label, value }));
+
+
+
 const Home: React.FC = () => {
 	const { replace } = useHistory();
 	const { organizationBaseInfo, activeOrganization, activeStudyField, setActiveOrganization, setActiveStudyField } = useAuth();
-
-	if(organizationBaseInfo?.length === 1 && organizationBaseInfo[0].fields.length === 1){
-		//There is only one organization, and one studyField, user has no buisness here.
-		//Mark his organization and studyField then move him away.
-		setActiveOrganization(organizationBaseInfo[0]);
-		setActiveStudyField(organizationBaseInfo[0].fields[0]);
-		replace("/questions");
-	}
-	const organizations = (organizationBaseInfo || []).map(({ name: label, id: value }) => ({ label, value }));
-	const relevantFields = (activeOrganization?.fields.map(({ name: label, id: value }) => ({ label, value })) || []);
+	const organizations = buildItemsForSelect(organizationBaseInfo || []); //user organizations
+	const relevantFields = buildItemsForSelect(activeOrganization?.fields || []); //organization fields
 
 	useEffect(() => {
 		if (organizationBaseInfo?.length === 1) {
 			setActiveOrganization(organizationBaseInfo[0])
+			if (organizationBaseInfo[0].fields.length === 1) {
+				//There is only one organization, and one studyField, user has no buisness here.
+				//Mark his organization and studyField then move him away.
+				setActiveStudyField(organizationBaseInfo[0].fields[0]);
+				replace("/questions");
+			}
 		}
-	}, [organizationBaseInfo, setActiveOrganization]);
+	}, [organizationBaseInfo, setActiveOrganization, setActiveStudyField, replace]);
+
+	const getSelectedIndex = (e: React.ChangeEvent<HTMLSelectElement>) => e.target.selectedIndex - 1;
+	const onOrganizationChanged = (e: React.ChangeEvent<HTMLSelectElement>) => {
+		const index = getSelectedIndex(e);
+		const organization = organizationBaseInfo?.find(org => org.id === organizations[index].value);
+		setActiveOrganization(organization);
+	};
+	const onStudyFieldChanged = (e: React.ChangeEvent<HTMLSelectElement>) => {
+		const index = getSelectedIndex(e);
+		const field = activeOrganization?.fields.find(f => f.id === relevantFields[index].value);
+		setActiveStudyField(field);
+	};
 
+	const handleClick = () => {
+		replace(`/${activeOrganization?.id}/${activeStudyField?.id}/questions`);
```

Comment:
I am thinking the `/${activeOrganization?.id}/${activeStudyField?.id}/` should move to a function in the auth hook, since it is used widely in depedents of the hook. What do you think?